### PR TITLE
Add JSON headers to Function generator template

### PR DESCRIPTION
**Pull request by ryancwalsh**
See @dthyresson suggestion: https://github.com/redwoodjs/redwoodjs.com/pull/682#issuecomment-832704119

### CODE REVIEW

**Comment by dthyresson** [comment]
@ryancwalsh Looks like the test snapshot needs updating.

```terminal
@redwoodjs/cli:  FAIL  src/commands/generate/function/__tests__/function.test.ts
@redwoodjs/cli:   ● creates a single word function file
@redwoodjs/cli:     expect(received).toMatchSnapshot()
@redwoodjs/cli:     Snapshot name: `creates a single word function file 1`
@redwoodjs/cli:     - Snapshot  - 0
@redwoodjs/cli:     + Received  + 4
@redwoodjs/cli:     @@ -19,10 +19,14 @@
@redwoodjs/cli:       export const handler = async (event, context) => {
@redwoodjs/cli:         logger.info('Invoked foo function')
@redwoodjs/cli:         return {
@redwoodjs/cli:           statusCode: 200,
@redwoodjs/cli:     +     headers: {
@redwoodjs/cli:     +       'Content-Type': 'application/json',
@redwoodjs/cli:     +     },
@redwoodjs/cli:     +
@redwoodjs/cli:           body: JSON.stringify({
@redwoodjs/cli:             data: 'foo function',
@redwoodjs/cli:           }),
@redwoodjs/cli:         }
@redwoodjs/cli:       }
```

Please try running the tests locally and making any changes to get the snapshot test back to green and happy. Thanks!

**Comment by ryancwalsh** [comment]
@dthyresson I'm having trouble. I don't know if it's because I'm on WSL. I've read https://github.com/redwoodjs/redwood/blob/main/CONTRIBUTING.md but am still stuck.

I cloned the Redwood framework to my local "redwood" directory and ran `yarn install`, and even before checking out the patch-1 branch, I'm still on "main", and neither `./tasks/test-tutorial` nor `yarn test` works as I'd expect (search for "fail" below):

```
rcwalsh (main) redwood $
rcwalsh (main) redwood $ ./tasks/test-tutorial
-bash: ./tasks/test-tutorial: No such file or directory
rcwalsh (main) redwood $ yarn test
yarn run v1.22.4
$ lerna run test --stream -- --colors --maxWorkers=4
lerna notice cli v3.22.1
lerna info Executing command in 12 packages: "yarn run test --colors --maxWorkers=4"
$ jest --colors --maxWorkers=4
$ jest --colors --maxWorkers=4
$ jest --colors --maxWorkers=4
@redwoodjs/internal:  PASS  src/__tests__/paths.test.ts
@redwoodjs/internal:  PASS  src/__tests__/config.test.ts
@redwoodjs/internal: Test Suites: 2 passed, 2 total
@redwoodjs/internal: Tests:       6 passed, 6 total
@redwoodjs/internal: Snapshots:   1 passed, 1 total
@redwoodjs/internal: Time:        4.8 s
@redwoodjs/internal: Ran all test suites.
$ jest --colors --maxWorkers=4
$ jest --colors --maxWorkers=4
@redwoodjs/auth:  PASS  src/__tests__/AuthProvider.test.tsx
@redwoodjs/auth:   ✓ Authentication flow (logged out -> login -> logged in -> logout) works as expected (97 ms)
@redwoodjs/auth:   ✓ Fetching the current user can be skipped (13 ms)
@redwoodjs/auth:   ✓ A user can be reauthenticated to update the "auth state" (31 ms)
@redwoodjs/auth:   ✓ When the current user cannot be fetched the user is not authenticated (8 ms)
@redwoodjs/auth:   ✓ Authenticated user has assigned role access as expected (19 ms)
@redwoodjs/auth:   ✓ Authenticated user has not been assigned role access as expected (16 ms)
@redwoodjs/auth:   ✓ Authenticated user has not been assigned some role access but not others as expected (23 ms)
@redwoodjs/auth:   ✓ Authenticated user has assigned role access as expected (19 ms)
@redwoodjs/auth:   ✓ Authenticated user has assigned role access as expected (22 ms)
@redwoodjs/auth:   ✓ Authenticated user has assigned role access as expected (18 ms)
@redwoodjs/auth:   ✓ Authenticated user has assigned role access as expected (20 ms)
@redwoodjs/auth: Test Suites: 1 passed, 1 total
@redwoodjs/auth: Tests:       11 passed, 11 total
@redwoodjs/auth: Snapshots:   0 total
@redwoodjs/auth: Time:        6.315 s
@redwoodjs/auth: Ran all test suites.
@redwoodjs/forms:  PASS  src/__tests__/form.test.tsx
@redwoodjs/forms:   Form
@redwoodjs/forms:     ✓ does not crash (76 ms)
@redwoodjs/forms:     ✓ calls onSubmit (47 ms)
@redwoodjs/forms:     ✓ renders and coerces user-supplied values (154 ms)
@redwoodjs/forms:     ✓ finds nested form fields to coerce (65 ms)
@redwoodjs/forms:     ✓ supports ref forwarding (6 ms)
@redwoodjs/forms:     ✓ lets users pass custom coercion functions (63 ms)
@redwoodjs/forms:     ✓ supports "dataType" prop on input fields with deprecation warning (11 ms)
@redwoodjs/forms:     ✓ handles int and float blank values gracefully with console warnings (59 ms)
@redwoodjs/forms:     ✓ handles datetime blank values gracefully with console warnings (63 ms)
@redwoodjs/forms:     ✓ input fields issue a console.warn if an invalid transformValue is set (6 ms)
@redwoodjs/forms:     ✓ for a TextAreaField with transformValue set to "Json", it automatically sets JSON validation.  Bad JSON case (59 ms)
@redwoodjs/forms:     ✓ for a TextAreaField with transformValue set to "Json", and a validation function custom set, it warns the developer upon a bad JSON submission (11 ms)
@redwoodjs/forms: Test Suites: 1 passed, 1 total
@redwoodjs/forms: Tests:       12 passed, 12 total
@redwoodjs/forms: Snapshots:   0 total
@redwoodjs/forms: Time:        6.434 s
@redwoodjs/forms: Ran all test suites.
$ jest --colors --maxWorkers=4
$ jest --colors --maxWorkers=4
$ jest --colors --maxWorkers=4
create-redwood-app:  PASS  __tests__/web-template.test.js (6.634 s)
create-redwood-app:   ✓ Web/src/App should contain App (3 ms)
create-redwood-app: Test Suites: 1 passed, 1 total
create-redwood-app: Tests:       1 passed, 1 total
create-redwood-app: Snapshots:   0 total
create-redwood-app: Time:        9.361 s
create-redwood-app: Ran all test suites.
@redwoodjs/structure:  PASS  src/x/__tests__/URL.test.ts (7.916 s)
@redwoodjs/structure:  FAIL  src/model/__tests__/model.test.ts
@redwoodjs/structure:   ● Test suite failed to run
@redwoodjs/structure:     Cannot find module '@redwoodjs/internal' from 'src/hosts.ts'
@redwoodjs/structure:     Require stack:
@redwoodjs/structure:       src/hosts.ts
@redwoodjs/structure:       src/model/__tests__/model.test.ts
@redwoodjs/structure:       2 | import glob from 'glob'
@redwoodjs/structure:       3 |
@redwoodjs/structure:     > 4 | import { Paths, getPaths } from '@redwoodjs/internal'
@redwoodjs/structure:         | ^
@redwoodjs/structure:       5 |
@redwoodjs/structure:       6 | import { lazy } from './x/decorators'
@redwoodjs/structure:       7 |
@redwoodjs/structure:       at Resolver.resolveModule (../../node_modules/jest-resolve/build/index.js:306:11)
@redwoodjs/structure:       at Object.<anonymous> (src/hosts.ts:4:1)
@redwoodjs/structure:  FAIL  src/outline/__tests__/outline.test.ts
@redwoodjs/structure:   ● Test suite failed to run
@redwoodjs/structure:     Cannot find module '@redwoodjs/internal' from 'src/hosts.ts'
@redwoodjs/structure:     Require stack:
@redwoodjs/structure:       src/hosts.ts
@redwoodjs/structure:       src/outline/__tests__/outline.test.ts
@redwoodjs/structure:       2 | import glob from 'glob'
@redwoodjs/structure:       3 |
@redwoodjs/structure:     > 4 | import { Paths, getPaths } from '@redwoodjs/internal'
@redwoodjs/structure:         | ^
@redwoodjs/structure:       5 |
@redwoodjs/structure:       6 | import { lazy } from './x/decorators'
@redwoodjs/structure:       7 |
@redwoodjs/structure:       at Resolver.resolveModule (../../node_modules/jest-resolve/build/index.js:306:11)
@redwoodjs/structure:       at Object.<anonymous> (src/hosts.ts:4:1)
@redwoodjs/api:  PASS  src/__tests__/beforeResolverSpec.test.js (6.756 s)
@redwoodjs/structure:  PASS  src/model/util/__tests__/advanced_path_parser.test.ts
@redwoodjs/api:  PASS  src/webhooks/webhooks.test.ts (6.962 s)
@redwoodjs/api:   ● Console
@redwoodjs/api:     console.warn
@redwoodjs/api:       With the SecretKeyVerifier verifier, your body isn't signed with a secret
@redwoodjs/api:
@redwoodjs/api:       23 |   return {
@redwoodjs/api:       24 |     sign: ({ secret }) => {
@redwoodjs/api:     > 25 |       console.warn(
@redwoodjs/api:          |               ^
@redwoodjs/api:       26 |         `With the SecretKeyVerifier verifier, your body isn't signed with a secret`
@redwoodjs/api:       27 |       )
@redwoodjs/api:       28 |       return secret
@redwoodjs/api:
@redwoodjs/api:       at sign (src/auth/verifiers/secretKeyVerifier.ts:25:15)
@redwoodjs/api:       at signPayload (src/webhooks/index.ts:157:10)
@redwoodjs/api:       at Object.<anonymous> (src/webhooks/webhooks.test.ts:131:27)
@redwoodjs/api:     console.warn
@redwoodjs/api:       With the SecretKeyVerifier verifier, your body isn't signed with a secret
@redwoodjs/api:
@redwoodjs/api:       23 |   return {
@redwoodjs/api:       24 |     sign: ({ secret }) => {
@redwoodjs/api:     > 25 |       console.warn(
@redwoodjs/api:          |               ^
@redwoodjs/api:       26 |         `With the SecretKeyVerifier verifier, your body isn't signed with a secret`
@redwoodjs/api:       27 |       )
@redwoodjs/api:       28 |       return secret
@redwoodjs/api:
@redwoodjs/api:       at sign (src/auth/verifiers/secretKeyVerifier.ts:25:15)
@redwoodjs/api:       at signPayload (src/webhooks/index.ts:157:10)
@redwoodjs/api:       at Object.<anonymous> (src/webhooks/webhooks.test.ts:140:27)
@redwoodjs/structure:  PASS  src/model/util/__tests__/process_env_diagnostics.test.ts (10.788 s)
@redwoodjs/structure:  PASS  src/x/__tests__/vscode-languageserver-types-x.test.ts (11.337 s)
@redwoodjs/api:  PASS  src/functions/parseJWT.test.ts
@redwoodjs/api:  PASS  src/logger/logger.test.ts (8.333 s)
@redwoodjs/api:   ● Console
@redwoodjs/api:     console.log
@redwoodjs/api:       [Wednesday, May 5th, 2021, 2:51:43 PM] INFO (4000 on RCWDESK): logged with pretty printing on
@redwoodjs/api:
@redwoodjs/api:       at Object.<anonymous> (src/logger/logger.test.ts:297:15)
@redwoodjs/router:  PASS  src/__tests__/util.test.ts (7.791 s)
@redwoodjs/structure:  PASS  src/x/__tests__/prisma.test.ts
@redwoodjs/structure: Test Suites: 2 failed, 5 passed, 7 total
@redwoodjs/structure: Tests:       15 passed, 15 total
@redwoodjs/structure: Snapshots:   0 total
@redwoodjs/structure: Time:        16.409 s
@redwoodjs/structure: Ran all test suites.
@redwoodjs/web:  PASS  src/components/FetchConfigProvider.test.tsx (9.17 s)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
lerna ERR! yarn run test --colors --maxWorkers=4 exited 1 in '@redwoodjs/structure'
lerna WARN complete Waiting for 3 child processes to exit. CTRL-C to exit immediately.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I'll see if I can work around it but figured you might want to know.

**Comment by Tobbe** [comment]
@ryancwalsh The snapshots that need updating are for the regular jest tests, not the test-tutorial stuff.
So try just running `yarn test` in the root of the RW framework directory and see if that works

**Comment by ryancwalsh** [comment]
@Tobbe I did run `yarn test`; it's the 4th line in my code pasted above in my previous comment. The fact that some of the tests failed on "main" branch makes me think I must have missed a step, but I'm not sure what. Thanks.

**Comment by Tobbe** [comment]
@ryancwalsh You're right. Sorry I didn't notice when I first looked :(

File: packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap [hunk 1]
```diff
@@ -24,6 +24,10 @@ export const handler = async (event, context) => {
 
   return {
     statusCode: 200,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+
```
Comment (dthyresson) [inline packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap:30]: Let's remove this newline to be consistent with formatting.

File: packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap [hunk 2]
```diff
@@ -89,6 +96,10 @@ export const handler = async (event, context) => {
...
```
Comment (dthyresson) [inline packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap:102]: Let's remove this newline to be consistent with formatting.

File: packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap [hunk 3]
```diff
@@ -121,6 +132,10 @@ export const handler = async (event, context) => {
...
```
Comment (dthyresson) [inline packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap:138]: Let's remove this newline to be consistent with formatting.

Or are these snaps auto-generated?

**Review by dthyresson — CHANGES_REQUESTED** [review]
There were some extra newlines in the snapshots - I wasn't sure if these were generated or not. If so, please ignore and let me know. Thanks!

File: packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap (hunk 3 above)
Comment (ryancwalsh) [reply packages/cli/src/commands/generate/function/__tests__/__snapshots__/function.test.ts.snap:138]: @dthyresson  Yes, they were auto-generated: I ran `yarn test -u`, which updated 4 snapshots automatically (which I wouldn't have known where to find or how to update). Thanks.

**Review by dthyresson — APPROVED** [review]

**Comment by thedavidprice** [comment]
Nice work, all! 

@dthyresson anything to add to future release notes for this one? doesn't seem so...

**Pull request merged**